fhevm/tee_bit.go

```diff
@@ -0,0 +1,145 @@
+package fhevm
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func teeShlRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	return doShiftOperationGeneric(environment, caller, input, runSpan, func(a, b, typ uint64) uint64 {
+		switch typ {
+		case 1:
+			return uint64((uint8(a) << uint8(b)))
+		case 2:
+			return uint64((uint8(a) << uint8(b)))
+		case 3:
+			return uint64((uint16(a) << uint16(b)))
+		case 4:
+			return uint64((uint32(a) << uint32(b)))
+		case 5:
+			return (a << b)
+		default:
+			return 0
+		}
+	}, "teeShlRun")
+}
+
+func teeShrRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	return doShiftOperationGeneric(environment, caller, input, runSpan, func(a, b, typ uint64) uint64 {
+		switch typ {
+		case 1:
+			return uint64((uint8(a) >> uint8(b)))
+		case 2:
+			return uint64((uint8(a) >> uint8(b)))
+		case 3:
+			return uint64((uint16(a) >> uint16(b)))
+		case 4:
+			return uint64((uint32(a) >> uint32(b)))
+		case 5:
+			return (a >> b)
+		default:
+			return 0
+		}
+	}, "teeShrRun")
+}
+
+func teeRotlRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	return doShiftOperationGeneric(environment, caller, input, runSpan, func(a, b, typ uint64) uint64 {
+		switch typ {
+		case 1:
+			return uint64((uint8(a) << uint8(b)) | (uint8(a) >> (uint8(4) - uint8(b)%4)))
+		case 2:
+			return uint64((uint8(a) << uint8(b)) | (uint8(a) >> (uint8(8) - uint8(b)%8)))
```

Comment:
I think type-casting makes the code hard to read.

would it be possible to create variables like:
```
a1, b1 := uint8(a), uint8(b)
```